### ISSUE TITLE
Feature: Remote Cache DNS Cache Duration

### ISSUE THREAD

**Issue by PatrickDuncan** [prometheus/prometheus#12484]
### Proposal

Hello!

### Problem

We ran into a DNS mishap where we changed one of our DNS entries to the wrong host. Our metrics eventually started failing and we had alert detection in place for that. 

The problem is that Prometheus cached the DNS entry for 30 minutes; Prometheus succeeded to remote write to the previous DNS IPs for the cache duration and the missing metrics detection was extremely delayed.

### Feature

It's not clear to me if the long Prometheus DNS cache is intentional or not, so this could be a bug. Our Kubernetes cluster's coreDNS DNS TTL is 30 seconds. Does Prometheus resolve the remote write host and cache the IPs for an extended period of time?

My feature request is to add a configuration to https://prometheus.io/docs/prometheus/latest/configuration/configuration/#remote_write that allows a user to define the DNS cache. The new configuration could also add other DNS-specific configurations.

**Comment by devrishi-das** [comment]
Hello,
We are also facing a similar issue. We are building a blue-green server migration automation. When the DNS has been mapped to the new instance, it resolves to the old instance when pinged from the EC2 Prometheus is running on. Due to this, we get downtime alerts, as the the prometheus instance resolves the DNS to the new endpoints, only after the old instance has been stopped. Is there a way we can flush the DNS cache?

**Comment by GregSharpe1** [comment]
Hello,

We seem to be encountering a similar issue. In our case, we're using a latency-based Route53 record for our remote-write receiver endpoint, with Prometheus running in Agent mode to push metrics to this endpoint.

During testing, we scaled down the regional endpoint, expecting Prometheus to switch over to another record under the latency-based configuration. However, instead of switching, we observed that the Prometheus agent continues to attempt connections using the previously cached IPs, even though a DNS resolution on the Prometheus agent shows it should be using the new set of IPs.

**Comment by bwplotka** [comment]
Hello from the Bug Scrub Meeting!

We glanced over [the code quickly](https://github.com/prometheus/prometheus/blob/main/storage/remote/client.go#L183) and as far we see we use a standard http DNS client, we don't cache anything. There was a recent change where [added round robin feature](https://github.com/prometheus/prometheus/pull/15329) which you may try, but it does not add/touch caching.

It's likely that it's a TCP keep alive for HTTP Remote Write connections. Since the previous connection still works, DNS change will not trigger a connection reset.

To sum up, it seems the DNS handling works as intended. Do you propose some advanced feature to handle DNS changes that would impact connection etc? 
In any case a reproduction unit test would be amazing for this!
